che-theia-java-extension/src/browser/navigation/find-implementers.ts

```diff
@@ -0,0 +1,206 @@
+import { CommandRegistry, CommandContribution, CommandHandler, Command, MenuModelRegistry, MenuContribution } from "@theia/core";
+import { ILanguageClient, TextDocumentPositionParams, ExecuteCommandRequest, TextDocumentIdentifier, SymbolKind } from "@theia/languages/lib/browser";
+import { LanguageClientProvider } from "@theia/languages/lib/browser/language-client-provider";
+import { EditorManager, EDITOR_CONTEXT_MENU } from "@theia/editor/lib/browser";
+import { FIND_IMPLEMENTERS_COMMAND } from "../che-ls-jdt-commands";
+import URI from "@theia/core/lib/common/uri";
+import { Range } from 'vscode-languageserver-types';
+import { JavaKeybindingContexts } from "../java-keybinding-contexts";
+
+export interface Implementers {
+    searchedElement: string,
+    implementers: ImplementationItem[]
+}
+
+export interface ImplementationItem {
+    name: string,
+    kind: number,
+    location: ImplementationItemLocation
+}
+
+export interface ImplementationItemLocation {
+    uri: string,
+    range: Range
+}
+
+@injectable()
+export class FindImplementers implements QuickOpenModel, CommandContribution, KeybindingContribution, MenuContribution, CommandHandler {
+
+    private items!: QuickOpenItem[];
+    private command: Command = {
+        id: 'java.command.implementation',
+        label: 'Java: Open Implementation(s)'
+    };
+    
+    constructor(
+        @inject(CommandRegistry) protected readonly commands: CommandRegistry,
+        @inject(KeybindingRegistry) protected readonly keybindings: KeybindingRegistry,
+        @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
+        @inject(LanguageClientProvider) protected readonly languageClientProvider: LanguageClientProvider,
+        @inject(EditorManager) protected readonly editorManager: EditorManager
+    ) { 
+    }
+
+    async execute() {
+        const client = await this.languageClientProvider.getLanguageClient("java");
+        if (client) {
+            const implementersResponse = await this.doRequestToimplementers(client);
+
+            if (implementersResponse.implementers.length === 1) {
+                const firstItem = implementersResponse.implementers[0];
+                this.editorManager.open(new URI(firstItem.location.uri), {
+                    mode: 'activate',
+                    selection: Range.create(firstItem.location.range.start, firstItem.location.range.end)
+                });
+            }
+
+            this.items = [];
+            for (const implItem of implementersResponse.implementers) {
+                const item = implItem as ImplementationItem;
+                this.items.push(new ImplementerQuickOpenItem(item.name, item.kind, item.location, this.editorManager));
+            }
+
+            const itemNotFoundMessage = "Found 0 implementations"; 
+            const itemFoundMessage = "Found " + implementersResponse.implementers.length + " implementation(s) for " + implementersResponse.searchedElement
```

Comment:
you may use string templates like `Found ${implementersResponse.implementers.length} implementations`